.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+        run: |
+          echo ${{ needs.PREPARE-job.outputs.runner_label}} 
+          echo "DOCKER_BUILDKIT=1" >> $GITHUB_ENV 
+
+      - name: Check and set docker cache location
+        id: docker_cache_check
+        run: |
+          CACHE_DIR="$HOME/runner/docker-cache"
+          echo "CACHE_DIR=$CACHE_DIR"
+
+          if [ -d "$CACHE_DIR" ]; then
+            echo "Cache directory exists."
+            CACHE_SIZE=$(du -sh "$CACHE_DIR" | cut -f1)
+            echo "Cache directory size: $CACHE_SIZE"
+          else
+            echo "Cache directory does not exist. Creating..."
+            sudo mkdir -p "$CACHE_DIR"
+            sudo chown $(whoami):$(id -gn) "$CACHE_DIR"
+            echo "Cache directory created."
+          fi
+
+          echo "HOME=$HOME"
+          echo "CACHELOC=$CACHE_DIR" >> $GITHUB_ENV
+          echo "CACHELOC is set to ${CACHE_DIR} ".           
+
+      - name: Build Docker image locally, save to tar file and move to persistent storage
+        uses: docker/build-push-action@v6
+        if: needs.PREPARE-job.outputs.devmode != 'true'
+        with:
+          context: ${{ github.workspace }}/${{ needs.PREPARE-job.outputs.directory }}
+          file: ${{ github.workspace }}/${{ needs.PREPARE-job.outputs.files }}
+          tags: | 
+            ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }}
+          platforms: ${{ needs.PREPARE-job.outputs.platform }}
+          push: false
+          load: true
+          provenance: false
+          cache-from: type=local,src=${{env.CACHELOC}}
+          cache-to: type=local,dest=${{env.CACHELOC}},mode=max
+          # cache-from: type=gha
+          # cache-to: type=gha,mode=max
+
+      - name: Build Docker image locally, for development mode
+        env:
+          CACHELOC: ${{ env.CACHELOC }}
+        if: needs.PREPARE-job.outputs.devmode == 'true'
+        run: |
+          docker buildx build \
+            --progress plain \
+            --cache-from type=local,src=${CACHELOC} \
+            --cache-to type=local,dest=${CACHELOC},mode=max \
+            --provenance=false \
+            --load \
+            --platform ${{ needs.PREPARE-job.outputs.platform }} \
+            --file ${{ github.workspace }}/${{ needs.PREPARE-job.outputs.files }} \
+            --tag ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }} \
+            ${{ github.workspace }}/${{ needs.PREPARE-job.outputs.directory }}
+
+      - name: Save Docker image to tar file
+        run: |
+          docker save -o ${GITHUB_WORKSPACE}/image.tar ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }}
+          echo "Docker image saved to ${GITHUB_WORKSPACE}/image.tar"
+
+      - name: Move image.tar to local persistent storage "$HOME/runner/artifacts"
```

Comment:
For really large OCI this might take a while. Do we want to add a `org.opencontainers.image.notar=true` functionality like we have for `noscan`?

Reply:
The use of the TAR file is aimed at faster data (image) transfer. As mentioned earlier, each job is stateless, and their communication typically relies on artifacts. However, uploading and downloading artifacts from GitHub for each job is too slow. We package the OCI image into a TAR file for two main reasons:

1. To transfer it more quickly to Trivy for scanning (although Trivy supports OCI, pulling via Docker is too slow).  
2. To pass it to Acacia, enabling rapid deployment on Ella or Setonix.

Without using TAR, these steps would require first uploading the packaged image to Quay.io, then downloading it for the scanning task (on the compilation machine), and finally downloading it again for Ella. By using TAR, we store the file on the local disk and then transfer it to Acacia and then download it from it for Ella, significantly speeding up the process. For example, a 10GB image now takes only 4 minutes to complete these steps, compared to nearly 1.5 hours without TAR.

Reply:
Therefore, this issue ultimately boils down to what I mentioned earlier: **precisely defining the tasks that should be done for each step in the workflow**. While I’m happy to add the notarization feature, it would likely require removing the `scanning `and `automated deployment` functionalities; otherwise, the process would become too slow.

With the current workflow using TAR, I can usually submit and deploy 20-30 versions per day. Without TAR, the process would be significantly slower and simply infeasible.

Reply:
Thanks for the explanation @Quleaf . Can we add this discussion in the README?

Reply:
Done. Thanks for your concerning. Also, I mark this issue as `to be discussed` in the github action readme file.